geopandas/io/geoarrow.py

```diff
@@ -425,6 +549,9 @@ def construct_shapely_array(arr: pa.Array, extension_name: str):
     with GeoArrow extension type.
 
     """
+    if isinstance(arr, pa.ExtensionArray):
+        arr = arr.storage
```

Comment:
`arr.storage` is guaranteed to always exist on `pa.ExtensionArray`? That's not added by geoarrow-pyarrow, right?